ui/appui-react/src/appui-react/widgets/Widget.tsx

```diff
@@ -51,10 +51,18 @@ export interface Widget {
    * It is not possible to disable the floating of a widget if `allowedPanels` is an empty array.
    */
   readonly canFloat?: boolean | CanFloatWidgetOptions;
-  /** Defaults to `Floating` if widget is not allowed to dock to any panels. Otherwise defaults to `Closed`.
-   * @note If `Hidden` the widget will be hidden after the layout is restored independently of saved layout state.
+  /** The default state of the widget.
+   * - Defaults to `Floating` if the widget is not allowed to dock to any panels.
+   * - Otherwise, defaults to `Closed`.
+   *
+   * @note If set to `Hidden`, the widget will be hidden after the layout is restored, independently of the saved layout state. Set `useSavedState` to `true` to disable this behavior.
+   * @note If set to `Unloaded`, the widget will be unloaded after the layout is restored, independently of the saved layout state. Set `useSavedState` to `true` to disable this behavior.
    */
   readonly defaultState?: WidgetState;
+  /** When enabled, the widget will always use the saved layout state. `defaultState` will only be used for the initial layout setup.
+   * This is useful for scenarios where the widget should only be hidden/unloaded initially, but is always restored to the last state.
+   */
+  readonly useSavedState?: boolean;
```

Comment:
This will satisfy our requirements as-is, so thanks for getting this done so quickly.

It looks like this property will only change the functionality for widgets with default state Hidden. What if the default state is something visible, let's say Floating, but `useSavedState` is set to false? Then it wouldn't honor that property and use its saved state anyway.

Reply:
That's correct I believe the confusion comes from the naming aspect, so alternative flag names are welcome..

Reply:
Ideal API would probably look similar to this:
```
defaultState?: WidgetState;
state?: WidgetState.Hidden | WidgetState.Unloaded; // used to Hide/Unload widgets between re-loads
```

Reply:
Alternatively, instead of a `boolean` we could change the flag to `undefined` or `true`?

Reply:
Well, I actually prefer the naming as you have it now, just that the functionality is a bit inconsistent. Ideally, I would say the state should _always_ be restored (even hidden) AND it always honors your new `useSavedState` property.

But I get that you don't want to alter pre-existing behavior, so maybe it could function like I mentioned, but default `useSavedState` to false in the hidden/unloaded `defaultState` scenario, and true otherwise. Then, it just needs to be documented in a comment and it should be ok.

Reply:
One more suggestion is to add `initialState` that would be used for initial setup only (which is what you need, although there would be a little bit of ambiguity w/ `defaultState`).

Reply:
Thanks, added additional docs to clarify [f87f0c5](https://github.com/iTwin/appui/pull/1210/commits/f87f0c5d8c49d2e917ecf72dd0cf661bb3a1da7d)